docs/examples/cmark_spec_0.30.yml

```diff
@@ -517,12 +552,18 @@ cases:
       children:
         - type: paragraph
           children:
-            - type: link
-              url: /f%C3%B6%C3%B6
-              title: föö
+            - type: linkReference
               children:
                 - type: text
                   value: foo
+              label: foo
+              identifier: foo
+              referenceType: shortcut
+        - type: definition
+          identifier: foo
+          label: foo
+          title: föö
+          url: /föö
```

Comment:
This highlights two key differences to the current spec:

1. URLs are not HTML escaped: this makes sense, since it only needs to be done when generating HTML
2. References are not yet resolved to definitions

(2) Is a more conceptual question: should this "snapshot" of the AST, represent it before or after reference resolution?

Reply:
There are certainly use cases (such as LSPs) where one would like to retain information regarding the position of definitions and references